### ISSUE TITLE
Unable to use the @MainActor initializers with nonisolated PhotosPicker in Xcode 16.0 RC with Swift 6 language mode

### ISSUE THREAD

**Issue by samsonjs** [SFSafeSymbols/SFSafeSymbols#134]
`PhotosPicker` declares its initializer as `@preconcurrency nonisolated` and its label closure is also `nonisolated`. This means that this code doesn't compile in Xcode 16.0 RC:

```swift
import PhotosUI
import SFSafeSymbols
import SwiftUI

struct ContentView: View {
    @State private var item: PhotosPickerItem?

    var body: some View {
        VStack {
            PhotosPicker(selection: $item) {
                Label("hello", systemSymbol: .antFill)
            }
        }
        .padding()
    }
}
```

![](https://github.com/user-attachments/assets/a91d9715-dc23-4a4b-be07-3c64d8742a3d)

**Comment by samsonjs** [comment]
I'm using this work-around in the meantime: https://gist.github.com/samsonjs/9a73531beb407b48a4a25e986c3dd01e

**Comment by samsonjs** [comment]
Filed as FB15150266 because maybe `PhotosPicker` is at fault here: https://gist.github.com/samsonjs/5c728655b877c9d3a0475d04e863ddd3

**Comment by samsonjs** [comment]
A better work-around is to define a nonisolated initializer and use that instead. e.g.

```swift
public extension Label where Title == Text, Icon == Image {
    nonisolated init(_ titleKey: LocalizedStringKey, systemSymbol: SFSymbol?) {
        self.init(titleKey, systemImage: systemSymbol?.rawValue ?? "")
    }
    
    @_disfavoredOverload
    nonisolated init<S>(_ title: S, systemSymbol: SFSymbol?) where S : StringProtocol {
        self.init(title, systemImage: systemSymbol?.rawValue ?? "")
    }
}
```

**Comment by samsonjs** [comment]
Looks like this is fixed in Xcode 16.1 beta 2 so I'll close this issue.
